server/pxf-json/src/test/java/org/greenplum/pxf/plugins/json/parser/JsonRecordReaderSeekNoSeekTest.java

```diff
@@ -0,0 +1,154 @@
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.greenplum.pxf.api.model.RequestContext;
+import org.greenplum.pxf.plugins.json.JsonRecordReader;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Comparator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class JsonRecordReaderSeekNoSeekTest {
+
+    private static final Log LOG = LogFactory.getLog(JsonRecordReaderSeekNoSeekTest.class);
+    private static final String RECORD_MEMBER_IDENTIFIER = "json.input.format.record.identifier";
+    private JobConf jobConf;
+    private String[] hosts = null;
+
+    @BeforeEach
+    public void setup() {
+        RequestContext context = new RequestContext();
+        context.setConfiguration(new Configuration());
+
+        jobConf = new JobConf(context.getConfiguration());
+        jobConf.set(RECORD_MEMBER_IDENTIFIER, "name");
+    }
+
+    @Test
+    public void testSeek() throws IOException {
+        File testsDir = new File("src/test/resources/parser-tests/seek");
+        File[] dirs = testsDir.listFiles();
+
+
+        for (File jsonDir : dirs) {
+            File childFile = new File(jsonDir, "input.json");
+
+            File[] jsonObjectFiles = jsonDir.listFiles(new FilenameFilter() {
+                public boolean accept(File file, String s) {
+                    return s.contains("expected");
+                }
+            });
```

Comment:
Putting this comment here on the first occurrence, but you can replace all occurrences of `FilenameFilter` with a lambda like this

```suggestion
            File[] jsonObjectFiles = jsonDir.listFiles((dir, name) -> name.contains("expected"));
```